src/pure.js

```diff
@@ -99,25 +130,91 @@ function render(
   }
 }
 
-function cleanup() {
-  mountedContainers.forEach(cleanupAtContainer)
+function render(
+  ui,
+  {
+    container,
+    baseElement = container,
+    legacyRoot = typeof ReactDOM.createRoot !== 'function',
+    queries,
+    hydrate = false,
+    wrapper,
+  } = {},
+) {
+  if (!baseElement) {
+    // default to document.body instead of documentElement to avoid output of potentially-large
+    // head elements (such as JSS style blocks) in debug output
+    baseElement = document.body
+  }
+  if (!container) {
+    container = baseElement.appendChild(document.createElement('div'))
+  }
+
+  let root
+  // eslint-disable-next-line no-negated-condition -- we want to map the evolution of this over time. The root is created first. Only later is it re-used so we don't want to read the case that happens later first.
```

Comment:
I think the rule is only complaining about the fact that this code could be written like this:

```tsx
if (mountedContainers.has(container)) {
  // the forEach here
} else {
  // create the root impl here.
}
```

Reply:
This was my interpretation as well. But I wanted to map the code to the lifetime of the root/container tuple as well: 
1. gets created
2. gets reused

ESLint is telling me to put the "gets reused" branch first which is not what happens in time. So if you're reading the code top-down you don't know where the root/container tuple is coming from. 

I should rewrite the comment since I tried to explain the reason there. 

Basically, I don't think `no-negated-condition` does anything useful. I could just as well write

```js
const needsRoot = !mountedContainers.has(container)
if (needsRoot) {
  // create the root impl here.
} else {
  // the forEach here
}
```
which means I have to name one more thing but then I don't have to use the noisy eslint-disable with confusing comments.